packages/web/components/bridge/immersive/immersive-bridge.tsx

```diff
@@ -0,0 +1,272 @@
+import { Transition } from "@headlessui/react";
+import { MinimalAsset } from "@osmosis-labs/types";
+import { isNil } from "@osmosis-labs/utils";
+import { memo, PropsWithChildren, useState } from "react";
+import { useLockBodyScroll } from "react-use";
+
+import { Icon } from "~/components/assets";
+import { AmountScreen } from "~/components/bridge/immersive/amount-screen";
+import { AssetSelectScreen } from "~/components/bridge/immersive/asset-select-screen";
+import { Screen, ScreenManager } from "~/components/screen-manager";
+import { StepProgress } from "~/components/stepper/progress-bar";
+import { Button, IconButton } from "~/components/ui/button";
+import { EventName } from "~/config";
+import { BridgeFlowProvider } from "~/hooks/bridge";
+import { useAmplitudeAnalytics } from "~/hooks/use-amplitude-analytics";
+import { useDisclosure } from "~/hooks/use-disclosure";
+import { FiatRampKey } from "~/integrations";
+import { ModalCloseButton } from "~/modals";
+import { FiatOnrampSelectionModal } from "~/modals/fiat-on-ramp-selection";
+import { FiatRampsModal } from "~/modals/fiat-ramps";
+import { api } from "~/utils/trpc";
+
+enum ImmersiveBridgeScreens {
```

Comment:
Since screen prop is a string key anyways, wouldn't it be slightly less verbose to just use string literal type here?

Reply:
While it's true, achieving type safety would otherwise be difficult and verbose. We'd have to supply the union type to every Screen component. I think an enum can solve this problem elegantly. What do you think?

Reply:
Since you're using string indexes, thoughts on using a const enum instead? Then, TS will remove it at bulid time and use indexes under the hood